packages/wrangler/src/bundle-reporter.tsx

```diff
@@ -21,9 +25,44 @@ export async function printBundleSize(
 ) {
 	const { size, gzipSize } = await getSize([...modules, main]);
 
-	const bundleReport = `${(size / 1024).toFixed(2)} KiB / gzip: ${(
-		gzipSize / 1024
+	const bundleReport = `${(size / ONE_KIB_BYTES).toFixed(2)} KiB / gzip: ${(
+		gzipSize / ONE_KIB_BYTES
 	).toFixed(2)} KiB`;
 
 	logger.log(`Total Upload: ${bundleReport}`);
+
+	if (gzipSize > ONE_MIB_BYTES && !process.env.NO_SCRIPT_SIZE_WARNING) {
+		logger.warn(
+			"We recommend keeping your script less than 1MiB (1024 KiB) after gzip. Exceeding past this can affect cold start time"
+		);
+	}
+}
+
+export function printOffendingDependencies(
```

Comment:
Might be something to move to the ESBuild files and export, keeping them co-located. (non-blocking) 

Thoughts @rozenmd ?

Reply:
yeah, unsure about these failures. some where definitely caused by OOMing node with the big buffers in the tests, but i clamped that down and seems to be better. just rebased off main so hopefully this works